packages/apollo-gateway/src/index.ts

```diff
@@ -145,24 +211,31 @@ export class ApolloGateway implements GraphQLService {
     config: GatewayConfig,
   ): Promise<[ServiceDefinition[], boolean]> {
     if (isLocalConfig(config)) return [config.localServiceList, false];
-    if (!config.serviceList)
-      throw new Error(
-        'The gateway requires a service list to be provided in the config',
-      );
-
-    const [
-      remoteServices,
-      isNewService,
-    ] = await getServiceDefinitionsFromRemoteEndpoint({
-      serviceList: config.serviceList,
-      ...(config.introspectionHeaders
-        ? { headers: config.introspectionHeaders }
-        : {}),
-    });
 
-    this.createServices(remoteServices);
+    const getServiceDefinitions = async () => {
+      if (isRemoteConfig(config)) {
+        return getServiceDefinitionsFromRemoteEndpoint({
+          serviceList: config.serviceList,
+          ...(config.introspectionHeaders
+            ? { headers: config.introspectionHeaders }
+            : {}),
+        });
+      } else {
+        if (!this.engineConfig) {
+          throw new Error(
+            'Must supply engineConfig to ApolloGateway#load() when no serviceList is provided.',
```

Comment:
This error _message_ might be confusing for someone not using TypeScript since they're unlikely to know what an `engineConfig` is, or it sounds like a property that might get passed to `load( { engineConfig: {} }`), though it's certainly good to have this error here.

Perhaps we're better off referring to the documentation?:

```
When `serviceList` is not set, an Apollo Engine configuration must be provided.  See
https://docs.url for more information.
```

Reply:
Put a link to docs that will be added with #2912.